src/execution/mod.rs

```diff
@@ -342,6 +344,260 @@ where
+                                                    let mut stack = Stack::new();
+                                                    // TODO: fully implement run_const
+                                                    run_const(wasm, &mut stack, (), &[]);
+                                                    let value = stack.peek_unknown_value();
+                                                    if value.is_none() {
+                                                        panic!("No value on the stack for element segment offset");
+                                                    }
+                                                    value.unwrap()
+                                                };
+
+                                                // TODO: this shouldn't be a simple value, should it? I mean it can't be, but it can also be any type of ValType
+                                                // TODO: also, do we need to forcefully make it i32?
+                                                let offset: u32 = match value {
+                                                    Value::I32(val) => val,
+                                                    Value::I64(val) => {
+                                                        if val > u32::MAX as u64 {
+                                                            panic!("i64 value for data segment offset is out of reach")
+                                                        }
+                                                        val as u32
+                                                    }
+                                                    // INFO: no need to implement all of them, it's either i32 or i64, otherwise offset is WRONG
+                                                    // INFO2: wait, we might need globals handling, now that I think about it, but make it return an u32 anyways
+                                                    _ => unreachable!(),
+                                                };
+                                                let offset: usize = offset as usize;
+                                                Ref::Extern(ExternAddr::new(Some(offset)))
+                                            }).collect::<Vec<Ref>>()
+                                        },
+                                        ElemItems::RefFuncs(_) => unreachable!()
+                                    }
+                                }
+                            },
+                        };
+                        let table = &mut tables[table_idx];
+
+                        assert!(table.len() >= (offset + el.len()));
+
+                        el.elem.iter().enumerate().for_each(|(i, rref)| {
+                            table.elem[i + offset] = *rref;
+                        });
+
+                        Some(el)
+                    }
+                    ElemMode::Declarative => {
```

Comment:
Declarative Segments are for streaming compilers only. As we don't do that, we can safely ignore them when executing.
WARNING: we can not fully get rid of them, we need to keep track of all elements, and if we don't, we're gonna break